onnxscript/_internal/converter.py

```diff
@@ -1117,6 +1131,7 @@ def ret(exp, i, suffix):
             preferred_name = f"return_val{suffix}"
             return_var = self._translate_expr(exp, preferred_name)  # TODO(rama)
             val = self._lookup(return_var.name, self._source_of(exp), False)
+            # TODO(rama): Can we avoid using val.kind here? And rely on the ir.Value methods like is_graph_input()?
```

Comment:
@justinchuby : I was looking into eliminating and simplifying some of the extra information currently tracked by the converter. One of this is this values.DynamicKind ... I remember that in some other work I was doing earlier, I found out that methods like is_graph_input on ir.Value works fine when we deserialize a model. But when constructing/building an IR from scratch, it seemed trickier to use, since it was set at some later stage of the construction. So: when is it safe to use those methods? I guess if it has been added as an input to some graph, then it is guaranteed to return true?

Reply:
As long as the value is added to the graph input/output list, the methods will work